benchmarks/servlets/src/main/java/org/eclipse/jetty/servlets/CrossOriginFilterBenchmark.java

```diff
@@ -0,0 +1,152 @@
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+public class CrossOriginFilterBenchmark 
+{
+    // cases to use for the benchmark
+    @SuppressWarnings("unchecked")
+    private static final List<String> [] cases = new List[] 
+    {
+        // one method
+/* 0 */ Arrays.asList("GET"),
+        // two methods
+/* 1 */ Arrays.asList("GET", "POST"),
+        // three methods, this is the default
+/* 2 */ Arrays.asList("GET", "POST", "HEAD"),
+        // this is another configuracion pausible 
+/* 3 */ Arrays.asList("GET", "POST", "HEAD", "PUT","DELETE"),
+        // this is an insame configuration, every registered methods, 2017-04-14 version of
+        // https://www.iana.org/assignments/http-methods/http-methods.xhtml 
+/* 4 */ Arrays.asList("ACL", "BASELINE-CONTROL", "BIND", 
+        "CHECKIN", "CHECKOUT", "CONNECT", "COPY", "DELETE", "GET", "HEAD", "LABEL", "LINK", "LOCK", 
+        "MERGE", "MKACTIVITY", "MKCALENDAR", "MKCOL", "MKREDIRECTREF", "MKWORKSPACE", "MOVE", "OPTIONS", 
+        "ORDERPATCH", "PATCH", "POST", "PRI", "PROPFIND", "PROPPATCH", "PUT", "REBIND", "REPORT", "SEARCH", 
+        "TRACE", "UNBIND", "UNCHECKOUT", "UNLINK", "UNLOCK", "UPDATE", "UPDATEREDIRECTREF", "VERSION-CONTROL"),
+    };
+    
+    // pattern version of the cases
+    private static final Pattern[]   PATTERNS = Stream.of(cases)
+                                                     .map(CrossOriginFilter::patternize)
+                                                     .toArray(Pattern[]::new);
+    
+    // method to use for the HIT benchmark: the last one that the user wrote
+    private static final String[]  HIT_METHOD = Stream.of(cases).map( l -> l.get(l.size() -1))
+                                                     .toArray(String[]::new);
+    
+    // method to use in MISS benchmark
+    private static final String MISS_METHOD   = "FOO";
+    
+    @Param({"0", "1", "2", "3", "4"})
+    public int arg;
+
+    @SuppressWarnings("unchecked")
+    private static final Collection<String>[] TREE_SETS = Stream.of(cases)
+                                                     .map(TreeSet::new)
+                                                     .toArray(Collection[]::new);
+
+    @SuppressWarnings("unchecked")
+    private static final Collection<String>[] HASHSET_SETS = Stream.of(cases)
+                                                     .map(HashSet::new)
+                                                     .toArray(Collection[]::new);
+
+    
+    @Benchmark
+    public boolean pattern_hit()  
+    {
+        return CrossOriginFilter.isMethodAllowed(PATTERNS[arg], HIT_METHOD[arg]);
+    }
+    
+    @Benchmark
+    public boolean pattern_miss()  
+    {
+        return CrossOriginFilter.isMethodAllowed(PATTERNS[arg], MISS_METHOD);
+    }
+    
+    @Benchmark
+    public boolean contains_hit()  
+    {
+        return cases[arg].contains(HIT_METHOD[arg]);
+    }
+    
+    public boolean contains_hits()  
```

Comment:
This one has no annotation and is a copy of the above method. Some method names end with "hit" , others with "hits".